### PR TITLE
KAA-326: Update Bootstrap extension implementation for Kaa C++ SDK.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/client-multi/client-cpp/impl/bootstrap/BootstrapManager.cpp [hunk 1]
```diff
@@ -114,15 +79,17 @@ void BootstrapManager::useNextOperationsServer(ChannelType type)
     }
 }
 
-void BootstrapManager::useNextOperationsServerByDnsName(const std::string& name)
+void BootstrapManager::useNextOperationsServerByAccessPointId(const std::int32_t& id)
```
Comment (izeigerman) [inline client/client-multi/client-cpp/impl/bootstrap/BootstrapManager.cpp]: Don you really need const reference here?

File: client/client-multi/client-cpp/impl/KaaClient.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 #include "kaa/KaaClient.hpp"
 
+#include "../kaa/channel/connectivity/IPConnectivityChecker.hpp"
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/KaaClient.cpp]: Need to specify absolute path.

File: client/client-multi/client-cpp/impl/channel/connectivity/IPConnectivityChecker.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#include "kaa/channel/TransportProtocolIdConstants.hpp"
+
+namespace kaa {
+
+bool IPConnectivityChecker::checkConnectivity()
+{
+    try {
+        ITransportConnectionInfoPtr server = serverStorage_.getPingServer();
+
+        if (server || server->getTransportId() == TransportProtocolIdConstants::HTTP_TRANSPORT_ID ||
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/channel/connectivity/IPConnectivityChecker.cpp]: Need logical AND in the first check.

File: client/client-multi/client-cpp/impl/event/EventManager.cpp [hunk 4]
```diff
@@ -65,7 +65,7 @@ void EventManager::produceEvent(const std::string& fqn
         event.target.set_string(target);
     }
 
-    if (trxId.get() != nullptr) {
+    if (trxId.get()) {
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/event/EventManager.cpp]: I propose to use an implicit conversation to the bool type.

File: client/client-multi/client-cpp/impl/event/EventManager.cpp [hunk 5]
```diff
@@ -140,9 +164,12 @@ void EventManager::onEventListenersReceived(const EventSyncResponse::eventListen
         const auto& container = listenersResponses.get_array();
 
         for (const auto& response : container) {
+            KAA_MUTEX_UNIQUE_DECLARE(lock, eventListenersGuard_);
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/event/EventManager.cpp]: I think the lock creation may be moved out of the cycle.

File: client/client-multi/client-cpp/impl/bootstrap/BootstrapManager.cpp (hunk 1 above)
Comment (ghost) [inline client/client-multi/client-cpp/impl/bootstrap/BootstrapManager.cpp]: Oh, my fault.

**Comment by andreypanasenko** [comment]
LGTM

**Pull request merged**